katana/katana.py

```diff
@@ -90,6 +93,207 @@ def readconfig():
     cfg = json.loads("".join(lines))
     return cfg
 
+@route('/searchkw', method='POST')
+def searchkw():
+    """
+    This method returns the ActionFile Path of the selected Driver.
+    """
+    value = parseString("".join(request.body))
+    indented_xml = "".join(value.toprettyxml(newl='\n'))
+    corrected_xml = remove_extra_newlines_char_xml(indented_xml)
+    corrected_xml = corrected_xml
+    with open('obj.txt', 'w') as f:
+        f.write(corrected_xml)
+    tree = ET.parse('obj.txt')
+    root = tree.getroot()
+    driver_name = root.text
+    global gpysrcdir
+    driver_file = gpysrcdir + '/ProductDrivers/' + driver_name + ".py"
+    actiondir_new = mkactiondirs(driver_file)
+    py_files = mkactionpyfiles(actiondir_new)
+    return py_files
+
+
+@route('/parsexmlobj', method='POST')
+def parsexmlobj():
+    """
+    This method fetch the XML object from the Katana UI of Keyword sequencing tool screen.
+    And parse it to form a new Wrapper keyword & place it in the user provided file path.
+    """
+    method_names = []
+    arg_name_list = ""
+    object_list_new = []
+    class_list_new = []
+    import_action_list = []
+    sample = []
+    py_files = ""
+    doc_string_value = []
+    kw_list_1 = []
+    xmlobj = parseString("".join(request.body))
+    indented_xml = "".join(xmlobj.toprettyxml(newl='\n'))
+    corrected_xml = remove_extra_newlines_char_xml(indented_xml)
+    with open('output.txt', 'w') as f:
+        f.write(corrected_xml)
+    tree = ET.parse('output.txt')
+    root = tree.getroot()
+    WrapperName = getChildTextbyParentTag('output.txt',
+                                          'Details',
+                                          'WrapperName')
+
+    ActionFile = getChildTextbyParentTag('output.txt', 'Details', 'ActionFile')
+    Description = getChildTextbyParentTag('output.txt', 'Details',
+                                          'Description')
+
+    # To check if the user provided wrapper keyword name already exists in action file
+    ActionFile = ActionFile.strip()
+    with open(ActionFile, 'r') as files:
+        for line in files:
+            line = line.split()
+            if "def" in line:
+                next_word = line[line.index("def")+1].split("(")[0]
+                method_names.append(next_word)
+
+    if WrapperName in method_names:
+        checkval = "yes"
+    else:
+        checkval = "no"
```

Comment:
Please either document this block of code or split it into different functions

Reply:
Done. Added comments for each block